### PR TITLE
refactor(validation): move request validations to RPC handlers

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Move RPC request validation into the RPC handlers instead of as a top-level middleware on the standalone server.

This change allows developers who use OpenFGA as a library to get the same validations that the standalone server provides. Since middleware doesn't apply to the RPC handler, we move validation inside of the RPC handler.

## References
- Closes https://github.com/openfga/openfga/issues/965
- Relates to https://github.com/openfga/cli/issues/141 - this PR allows the CLI test command to have proper validations when the OpenFGA server is embedded
- https://github.com/openfga/openfga/issues/871

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -318,6 +320,10 @@ func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequ
 	))
 	defer span.End()
 
+	if err := req.Validate(); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
```
Comment (miparnisari) [inline pkg/server/server.go:325]: Do you mind sharing a screenshot showing that the http status code of a validation error hasn't changed?

Comment (jon-whit) [reply pkg/server/server.go:325]: Yeah I'll get one here in a sec, but the fact none of the tests failed should be a good indicator no?

Comment (jon-whit) [reply pkg/server/server.go:325]: And the HTTP status code mapping still happens in the grpc-gateway middleware, which applies to the standalone server. So as long as the `codes.InvalidArgument` is the same as it was before we gucci.

Comment (jon-whit) [reply pkg/server/server.go:325]: Here's a sample screenshot. 
Screenshot 2023-08-30 at 1 52 26 PM

The HTTP status code is 400 👍

**Review by miparnisari — APPROVED** [review]

**Pull request merged**